pkg/util/net/grpc/grpc.go

```diff
@@ -34,32 +35,33 @@ import (
 
 const (
 	CONNECT_TIMEOUT_SECONDS = 2
+	unixProtocol            = "unix"
 )
 
 func DialSocket(socketPath string) (*grpc.ClientConn, error) {
 	return DialSocketWithTimeout(socketPath, 0)
 }
 
 func DialSocketWithTimeout(socketPath string, timeout int) (*grpc.ClientConn, error) {
+	dialer := func(ctx context.Context, addr string) (net.Conn, error) {
+		return (&net.Dialer{}).DialContext(ctx, unixProtocol, addr)
+	}
 
 	options := []grpc.DialOption{
 		grpc.WithAuthority("localhost"),
-		grpc.WithInsecure(),
-		grpc.WithDialer(func(addr string, timeout time.Duration) (net.Conn, error) {
-			return net.DialTimeout("unix", addr, timeout)
-		}),
+		grpc.WithTransportCredentials(insecure.NewCredentials()),
+		grpc.WithContextDialer(dialer),
 		grpc.WithBlock(), // dial sync in order to catch errors early
 	}
 
+	ctxTimeout := time.Duration(timeout+CONNECT_TIMEOUT_SECONDS) * time.Second
```

Comment:
nit: can we rename the `CONNECT_TIMEOUT_SECONDS` to `defaultTimeoutSeconds` or anything similar?